R/R.sublime-syntax

```diff
@@ -187,123 +238,188 @@ contexts:
-          pop: true
+          pop: 1
+
     - match: (R|r)('(-*)\{)
       captures:
         1: storage.type.string.r
         2: punctuation.definition.string.begin.r
       push:
-        - meta_scope: string.quoted.single.r
+        - meta_include_prototype: false
+        - meta_scope: meta.string.r string.quoted.single.r
         - match: \}(\3)'
           scope: punctuation.definition.string.end.r
-          pop: true
+          pop: 1
+
     - match: (R|r)('(-*)\[)
       captures:
         1: storage.type.string.r
         2: punctuation.definition.string.begin.r
       push:
-        - meta_scope: string.quoted.single.r
+        - meta_include_prototype: false
+        - meta_scope: meta.string.r string.quoted.single.r
         - match: \](\3)'
           scope: punctuation.definition.string.end.r
-          pop: true
+          pop: 1
+
     - match: '"'
       scope: punctuation.definition.string.begin.r
       push:
-        - meta_scope: string.quoted.double.r
-        - match: \\.
-          scope: constant.character.escape.r
+        - meta_include_prototype: false
+        - meta_scope: meta.string.r string.quoted.double.r
+        - include: double-quoted-string-escapes
         - match: '"'
           scope: punctuation.definition.string.end.r
-          pop: true
+          pop: 1
+
     - match: "'"
       scope: punctuation.definition.string.begin.r
       push:
-        - meta_scope: string.quoted.single.r
-        - match: \\.
-          scope: constant.character.escape.r
+        - meta_include_prototype: false
+        - meta_scope: meta.string.r string.quoted.single.r
+        - include: single-quoted-string-escapes
         - match: "'"
           scope: punctuation.definition.string.end.r
-          pop: true
+          pop: 1
+
+  single-quoted-string-escapes:
+    - meta_include_prototype: false
+    - match: (\\)\'
+      scope: constant.character.escape.r
+      captures:
+        1: punctuation.definition.string.escape.r
+    - include: string-escapes
+
+  double-quoted-string-escapes:
+    - meta_include_prototype: false
+    - match: (\\)\"
+      scope: constant.character.escape.r
+      captures:
+        1: punctuation.definition.string.escape.r
+    - include: string-escapes
+
+  string-escapes:
+    - meta_include_prototype: false
+    - include: invalid-string-escapes
+    - include: valid-string-escapes
+
+  invalid-string-escapes:
+    - meta_include_prototype: false
+    - match: '(\\)0'                                        # null character
+      scope: invalid.illegal.constant-character-escape.r
+
+  valid-string-escapes:
+    - meta_include_prototype: false
+    - match: '\\[\\nrtbafv]'
+      scope: constant.character.escape.r
+    - match: '\\[0-7]{1,3}'
+      scope: constant.character.escape.octal.r
+    - match: '\\x\h{1,2}'
+      scope: constant.character.escape.hex.r
+    - match: '\\u\{?\h{1,4}\}?'
+      scope: constant.character.escape.unicode.r
```

Comment:
With regards to the `true/false/null` debate, maybe enable distinction of unicode escapes as already applied in python and some others?

```suggestion
      scope: constant.character.escape.unicode.16bit.r
```